src/main/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java

```diff
@@ -19,27 +15,26 @@
 import com.marginallyclever.makelangelo.turtle.TurtleMove;
 
 public class SaveGCode {
-	private JFileChooser fc = new JFileChooser();
-	private FileNameExtensionFilter filter = new FileNameExtensionFilter("GCode", "gcode");
-	
+	private final JFileChooser fc = new JFileChooser();
+
 	public SaveGCode() {
+		FileNameExtensionFilter filter = new FileNameExtensionFilter("GCode", "gcode");
```

Comment:
why is this recreated on every save?  to reduce memory?

Reply:
it is a small object only used in this method. it was declared as an instance field of the class and only used some times, no memory optimization but I like short live object.